server/src/main/java/org/opensearch/index/fielddata/IndexNumericFieldData.java

```diff
@@ -133,8 +134,16 @@ public final SortField sortField(
             : SortedNumericSelector.Type.MIN;
         SortField sortField = new SortedNumericSortField(getFieldName(), getNumericType().sortFieldType, reverse, selectorType);
         sortField.setMissingValue(source.missingObject(missingValue, reverse));
+        
+        // For long, doubles & dates, we can use this numeric sort optimiser.
```

Comment:
The comment should explain _why_ this is the case, instead of stating what the code does. I think the relevant bit here is that these types are the ones where SortField.Type matches the Point type. Is that right?

Reply:
Added the comments. let me know if it looks good !